Feat/venue detailed view

nattsnb opened the pull request.

Commit:
fixed banner styling and displaying  conditions. added useLocalization to provide porper components display depending on path

Commit:
added config directory with bannerConfig.ts, getBannerConfig function in utils to provide the confid, updated Banner component to display the elemnt according to the provided config

Commit:
if provided filters in any other location than explore venue view, on submit will navigate to '/' path

Commit:
added basic view composition with placeholders, provided picture context, added useVenueDetailedView.ts

Commit:
added calendar for booking the venue and fixed styles in files

Commit:
updated venueDto, added ContactInfo section

Commit:
added DetailsAndImageContainer, added fixed number of reviews

Commit:
added LinkBarAndBody component

nattsnb committed:
style(SearchBar/DataInput): removed unnecessary argument

nattsnb committed:
style( DetailsAndImageContainer): use uuid instead of index for mapping. added uuid package
fix(VenueDetailedView) added component name instead of placeholder, must have changed conditions to provide venue exists in both cases

nattsnb committed:
feat(Gallery) added gallery component

nattsnb committed:
feat(Gallery) added gallery component

nattsnb committed:
fix(VenueDetailedView) provided activeVenue with context

nattsnb committed:
fix(Gallery.styled) deleted unneeded type
feature(ExploreVenuesViews & Banner) moved buttons to toggle drawers to banner component, deleted double displayed banner. fixed styling
fix(SearchBar.styled) deleted muted and unneeded code

nattsnb committed:
style(Calendar) moved full prise calculations to useCalendar.ts
feat(DetailsAndImageContainer) provided number of stars and reviews based on data from server
fix(types) updated types according to backend

nattsnb committed:
feat(theme) added new colors to background palette
fix(styles) fixed theme interpolation nad prop types

nattsnb committed:
fix(NavBar) deleted inline styling

nattsnb committed:
after merge

nattsnb committed:
fix(SearchBar) fixed styling

prettier

nattsnb committed:
bug(all) flickering app while starting
fix(Gallery) deleted doubled component mounting
fix(Gallery&Map) provided screen width with prop not to use useMediaQuery multiple times
fix(VenueDetailedView) deleted doubled component mounting
(todo) provide is hidden prop to proper components while in mobile view

nattsnb committed:
bug(all) flickering app while starting
feat(VenueSection) component that will allow to generate most of the components only once
fix(WideBodyLinkBarAndContentContainer) doesn't generate the components anymore
style(VenueDetailedView) used css to hide components

nattsnb committed:
fix(VenueDetailedView) fix the flickering issue - no dependency in useEffect, reordered the initial checks

prettier

nattsnb committed:
fix(VenueDetailedView) fix the flickering issue - no dependency in useEffect, reordered the initial checks

prettier

nattsnb committed:
fix(VenueDetailedView) added missing dependencies

nattsnb committed:
feat(VenueSections) added styled div to provide border in desktop view and central alignment in mobile view

nattsnb committed:
fix(Calendar) fixed props name spelling to avoid mui warnings

nattsnb committed:
feat(VenueDetailedView) added displayedSection to ActiveVenue context to provide easier acces to data

nattsnb committed:
feat(VenueDetailedView) added drawer that display calendar in mobile view
fix(Calendar) added extra styling for calendar to display properly

nattsnb committed:
fix(Layout) deleted unneeded import
fix(Gallery,Map) isMobile is taken from hook now
fix(VenueDetailedView) div fixed to bottom is over map now

nattsnb committed:
fix(all) changed variables names to isMobile when there's change in display dependent on it
fix(Gallery, Map, VenueSections) changed wrong condition in useMediaQuery hook

nattsnb committed:
feat(theme) added new color
fix(Calendar) fixed buttons and calendar styling

Solnick commented on src/components/VenueDetailedView/Calendar/Calendar.styled.ts:
```diff
@@ -0,0 +1,151 @@
+    border: 1px solid ${theme.palette.primary.border};
+    border-radius: 15px;
+    padding-left: ${theme.spacing(4)};
+        padding-top:  0;
+        padding-bottom:  0;
+        padding-right:  ${theme.spacing(4)};
```
format the code

nattsnb replied:
done

Solnick commented on src/components/VenueDetailedView/Calendar/index.tsx:
```diff
@@ -0,0 +1,130 @@
+  const { activeVenue } = useActiveVenue();
+
+  if (!activeVenue) {
+    return <></>;
```
```suggestion
    return null;
```

nattsnb replied:
done

nattsnb committed:
feat(theme) added new color

nattsnb committed:
fix(Calendar) fixed code formatting and changed empty fragment into null

nattsnb committed:
feat(VenueDetailedView) used IntersectionObserver to make narrowNavBar stick to the top of mobile view

nattsnb committed:
fix(all) the app returns null instead of empty fragment

nattsnb committed:
fix(Calendar) fix styling of "starts at" "ends at" buttons
fix(theme) deleted unneeded background color
fix(Banner) sort and filter buttons are now displayed only in one location

nattsnb committed:
fix(SearchBar) added FormController and used it for GuestsNumberInput

nattsnb committed:
fix(all) added form controller for all inputs in SearchBar

fix(DateInput) deleted unneeded useDateInput file

nattsnb committed:
merged branch feat/explore-venues-view

nattsnb committed:
fix(NavBar) changed styling of link in mobile view

nattsnb committed:
fix(NavBar) changed name of variable

nattsnb committed:
fix(FiltersDrawer) fixed handleDropClick function

fix(ResultsList) app displays 'No venues found' if venuesOnPage.length is 0, deleted console.logs

nattsnb committed:
fix(DateInput) buttons starts at and ends at are properly styled and they change their appearance depending on which date is to be picked

nattsnb committed:
fix(SearchBar) changed labels into placeholders in SearchBar inputs, styled I want to be more specific button,
feat(theme) provided inputs not to change color when focused on
feat(readme) added read me

nattsnb committed:
fix(LocalizationInput) autoComplete is off

nattsnb committed:
feat(BackButton) Back Button is separate element now

Solnick commented on src/components/Banner/SearchBar/index.tsx:
```diff
@@ -61,13 +63,61 @@ export const SearchBar = () => {
+                    />
+                  )}
+                </FormController>
+                <FormController name="venueTypeId">
+                  {(field) => (
+                    <VenueTypeInput
+                      value={
+                        typeof field.value === "number"
+                          ? field.value
+                          : undefined
+                      }
+                      onChange={field.onChange}
+                    />
+                  )}
+                </FormController>
+                <FormController name="guests">
+                  {(field) => (
+                    <GuestsNumberInput
+                      value={field.value as number | null | undefined}
+                      onChange={field.onChange}
+                    />
+                  )}
+                </FormController>
```
Something with the field type is wrong, let;s fix it together

nattsnb replied:
done

Solnick commented on src/components/VenueDetailedView/Description/index.tsx:
```diff
@@ -0,0 +1,66 @@
+      <StyledSectionContainer>
+        <StyledAmenitiesList>
+          {activeVenue.amenityToVenues.map(
+            (amenityToVenue: AmenityToVenueDto) => (
```
typing by inference can be utilized here

nattsnb replied:
done

Solnick commented on src/components/VenueDetailedView/Gallery/index.tsx:
```diff
@@ -0,0 +1,42 @@
+      <StyledGalleryContainer>
+        {activeVenue.images.map((imageURL, index) => (
+          <StyledImageContainer
+            key={uuidv4()}
```
this key will be changed with every rerender, we can't leave it like this

nattsnb replied:
done

Solnick commented on src/components/VenueDetailedView/VenueSections.tsx:
```diff
@@ -0,0 +1,71 @@
+  return (
+    <StyledSectionContainer>
+      <section
+        style={{
```
do not use inline styles

nattsnb replied:
done

nattsnb committed:
feat(VenueDetailedView) deleted inline styles

nattsnb committed:
feat(Description) deleted unneeded type

feat(Gallery) changed image key

nattsnb committed:
feat(FormController) changed to generic one

feat(SearchBar) deleted unneeded types
